rfd/0215-bot-details-web.md

```diff
@@ -0,0 +1,218 @@
+
+#### Bot info and config
+
+Shows basic details and configuration. All items are readonly. Date/time items have a hover state which shows a tooltip with the full date and time.
+
+![[0215-feature-info.png]]
+**Data source**
+``` yaml
+# Bot resource (calculated)
+kind: bot
+metadata:
+  name: robot
+spec:
+  max_session_ttl:
+    seconds: 43200
+status:
+  role_name: bot-robot # Links to role
+  user_name: bot-robot # Links to user
+
+# User resource
+kind: user
+metadata:
+  labels:
+    teleport.internal/bot: robot # Links to bot
+  name: bot-robot
+spec:
+  created_by:
+    time: "2025-06-02T11:25:13.238653583Z"
+    user:
+      name: nicholas.marais@goteleport.com
+  roles:
+  - bot-robot # Links to role
+  status:
+    is_locked: false
+    lock_expires: "0001-01-01T00:00:00Z"
+    locked_time: "0001-01-01T00:00:00Z"
+
+# Role resource
+kind: role
+metadata:
+  name: bot-robot
+```
+#### Join tokens
+Lists Join Tokens with a role of "Bot" and `bot_name` matching the bot being viewed. An overflow menu allow navigating to Zero Trust Access > Join Tokens. Clicking an item navigates to to the view/edit page for that token.
+
+![[0215-feature-join-tokens.png]]
+**Data source**
+``` yaml
+# Token resource
+kind: token
+metadata:
+  name: robot-github
+spec:
+  bot_name: robot
+  join_method: github
+  roles:
+  - Bot
+```
+#### Roles and traits
+Provides full lists of roles and traits (well-known and custom). Edit operations are provided for each for convenience, which open the page-wise edit modal with all editable fields available.
+
+![[0215-feature-roles-traits.png]]
+**Data source**
+``` yaml
+# Bot resource
+kind: bot
+metadata:
+  name: robot
+spec:
+  roles:
+  - file-browser-access
+  - mac.lan-ssh-access
+  traits:
+  - name: logins
+    values:
+    - nick.marais
+```
+#### Active instances
+Lists the most recent (max 10) instances for the bot, ordered most recent first. A "see more" action navigates to the bot instances page with a pre-populated search filter on the bot's name - this is an imperfect filter as it's a contains-text filter across all fields
+
+![[0215-feature-active-instances.png]]
+
+#### Edit roles, traits and max session duration (`max_session_ttl`)
+Shows a dialog where the user can add and/or remove assigned roles, add and/or remove traits (well-known or custom), and edit the configured max session duration in the form `43200s`, `30m` or `3h`. Allow all changes to be made in a single atomic transaction.
+
+TODO: wireframe for edit form
+
+#### Delete bot
+Deletes the bot after confirmation. Shows a loading indicator during the call to the api. On success, navigates to the bots list (`/web/bots`). On error, shows a message within the confirmation dialog.
+
+![[0215-feature-delete-bot.png]]
+#### Lock bot
+Locks the bot after confirmation. Shows a loading indicator during the call to the api. On success, removes the dialog but remains on the bot detail page. On error, shows a message within the confirmation dialog.
```

Comment:
This should also expose the `Message` and `TTL` fields that are on the regular lock slide-out panel. Might want to make this the same slide-out panel as the regular lock process for UI consistency?